### ISSUE TITLE
CLI hangs with "unexpected: --output is required" (is it related to NodejsFunction?)

### ISSUE THREAD

**Issue by th3r10n** [aws/aws-cdk#8921]
After adding a NodejsFunction construct to my stack and having succesfully 
[controller.zip](https://github.com/aws/aws-cdk/files/4883316/controller.zip)

### Environment

  - **CLI Version      :**
  - **Framework Version:**
  - **Node.js Version:** 
  - **OS               :**
  - **Language (Version):** 

### Other

--- 

This is :bug: Bug Report

**Comment by eladb** [comment]
Please run with `—verbose` and pasting your output?

 Copy @jogold

**Comment by th3r10n** [comment]
CDK toolkit version: 1.49.1 (build 7d6321f)
Command line arguments: {
  _: [ 'synth' ],
  v: 1,
  verbose: 1,
  e: true,
  exclusively: true,
  r: undefined,
  'role-arn': undefined,
  roleArn: undefined,
  b: true,
  o: '',
  output: '',
  s: true,
  'ignore-errors': false,
  ignoreErrors: false,
  json: false,
  j: false,
  ec2creds: undefined,
  i: undefined,
  'version-reporting': undefined,
  versionReporting: undefined,
  'path-metadata': true,
  pathMetadata: true,
  'asset-metadata': true,
  assetMetadata: true,
  staging: true,
  'no-color': false,
  noColor: false,
  fail: false,
  '$0': 'cdk'
}
cdk.json: {
  "app": "npx ts-node bin/controller.ts",
  "context": {
    "@aws-cdk/core:enableStackNameDuplicates": "true",
    "aws-cdk:enableDiffNoFail": "true"
  }
}
merged settings: {
  versionReporting: true,
  pathMetadata: true,
  output: '',
  app: 'npx ts-node bin/controller.ts',
  context: {
    '@aws-cdk/core:enableStackNameDuplicates': 'true',
    'aws-cdk:enableDiffNoFail': 'true'
  },
  tags: [],
  assetMetadata: true,
  toolkitBucket: {},
  staging: true
}
Determining whether we're on an EC2 instance.
Does not look like EC2 instance.
Toolkit stack: CDKToolkit
Setting "CDK_DEFAULT_REGION" environment variable to us-west-2
Resolving default credentials
Retrieved account ID 905871336375 from disk cache
Setting "CDK_DEFAULT_ACCOUNT" environment variable to 905871336375
context: {
  '@aws-cdk/core:enableStackNameDuplicates': 'true',
  'aws-cdk:enableDiffNoFail': 'true',
  'aws:cdk:enable-path-metadata': true,
  'aws:cdk:enable-asset-metadata': true
}
unexpected: --output is required
Error: unexpected: --output is required
    at Object.execProgram (/usr/local/lib/node_modules/aws-cdk/lib/api/cxapp/exec.ts:60:11)
    at CloudExecutable.doSynthesize (/usr/local/lib/node_modules/aws-cdk/lib/api/cxapp/cloud-executable.ts:66:24)
    at CloudExecutable.synthesize (/usr/local/lib/node_modules/aws-cdk/lib/api/cxapp/cloud-executable.ts:52:29)
    at CdkToolkit.selectStacksForDiff (/usr/local/lib/node_modules/aws-cdk/lib/cdk-toolkit.ts:398:22)
    at CdkToolkit.synth (/usr/local/lib/node_modules/aws-cdk/lib/cdk-toolkit.ts:298:20)
    at main (/usr/local/lib/node_modules/aws-cdk/bin/cdk.ts:279:16)
    at initCommandLine (/usr/local/lib/node_modules/aws-cdk/bin/cdk.ts:185:9)

**Comment by jogold** [comment]
@eladb the error seems to originate from here:

https://github.com/aws/aws-cdk/blob/ae2378cc2a537277025c9104bc43a4cc68318650/packages/aws-cdk/lib/api/cxapp/exec.ts#L59-L61

any idea what might cause this and how it could be related to the `NodejsFunction` construct?

**Comment by eladb** [comment]
For some reason, the `output` switch in the CLI is an empty string (`""`) instead of `cdk.out`. @th3r10n can you also include the exact command line you are executing? Also, can you try to pass `-o cdk.out` to the CLI?

**Comment by th3r10n** [comment]
@eladb,

The output of executing "cdk synth -verbose": [cdk-synth-verbose.txt](https://github.com/aws/aws-cdk/files/4894411/cdk-synth-verbose.txt)

The output of executing "cdk synth -verbose -o cdk.out": [cdk-synth-verbose-output.txt](https://github.com/aws/aws-cdk/files/4894410/cdk-synth-verbose-output.txt)

**Comment by jogold** [comment]
@th3r10n and you don't get the `unexpected: --output is required` error when removing the `NodejsFunction` construct?

**Comment by th3r10n** [comment]
@jogold ,

If I remove the NodejsFunction construct I get:

The output of executing "cdk synth -verbose": [cdk-synth-verbose-no-function.txt](https://github.com/aws/aws-cdk/files/4900225/cdk-synth-verbose-no-function.txt)

The output of executing "cdk synth -verbose -o cdk.out":  [cdk-synth-verbose-output-no-function.txt](https://github.com/aws/aws-cdk/files/4900226/cdk-synth-verbose-output-no-function.txt)

However, without the above mentioned construct, "cdk synth" works fine.

The GitHub repo with the code is available here: https://github.com/th3r10n/controller.git

**Comment by jogold** [comment]
> The GitHub repo with the code is available here: [th3r10n/controller.git](https://github.com/th3r10n/controller.git)

This is an empty project. I don't see any `NodejsFunction` in there. Do you have issues synthesizing or deploying it?

**Comment by th3r10n** [comment]
@jogold ,

My mistake, I've populated the repo now. 

I have problems to synthesize it.

**Comment by jogold** [comment]
> My mistake, I've populated the repo now.

The code currently doesn't reference any `NodejsFunction` constructs. Also it fails because you're pointing to a non-existing `lambda` directory in the `GetProductsHandler` function.

**Comment by th3r10n** [comment]
@jogold ,

The ```NodeJSFunction``` is in line 53 of controller-stack.ts(commented out to make it work). I've added the missing js files referenced in the code.

Be that as it may. I let the cdk synth/deploy commands just hang there, turns out they are returning successfully but only after a long time compared to when the ```NodeJSFunction``` construct is not present.

**Comment by jogold** [comment]
@th3r10n just pulled the latest version of your repo.

```bash
$ time npm run cdk synth
```


![image](https://user-images.githubusercontent.com/12623249/87398688-8be54680-c5b6-11ea-887e-4989cc17627b.png)

**Comment by eladb** [comment]
What's the status here @jogold ?

**Comment by jogold** [comment]
> What's the status here @jogold ?

I was able to successfully synthesize...

> turns out they are returning successfully but only after a long time compared to when the NodeJSFunction construct is not present.

@th3r10n how long? only on the first run? The first run could take more time if the bundling Docker image is not already built and cached.

**Comment by th3r10n** [comment]
@jogold ,

If the NodeJSFunction is present, its taking around the ball park of 30 s not withstanding wether it's the first or a subsequent run:

<img width="152" alt="with" src="https://user-images.githubusercontent.com/8766032/87899340-3d590180-ca27-11ea-9de8-236bdff824cb.png">

Without it, it's way less:

<img width="159" alt="without" src="https://user-images.githubusercontent.com/8766032/87899361-506bd180-ca27-11ea-8e3e-90e203407fce.png">

**Comment by jogold** [comment]
@th3r10n are you on MacOS?

**Comment by th3r10n** [comment]
Yes @jogold. I'm on macOS.

**Comment by jogold** [comment]
OK, this is a duplicate of #9120

**Comment by eladb** [comment]
Dup #9120
